db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/SchedulerClient.java

```diff
@@ -36,18 +36,75 @@
 
 public interface SchedulerClient {
 
+    /**
+     * Schedule a new task based on a defined generic T type with a specific execution time
+     *
+     * @param taskInstance  Task instance with its data
+     * @param executionTime Execution date time
+     * @return void
+     * @see java.time.Instant
+     * @see com.github.kagkarlsson.scheduler.task.TaskInstance
+     */
     <T> void schedule(TaskInstance<T> taskInstance, Instant executionTime);
 
+    /**
+     * Schedule a new execution date to a already existent task
+     *
+     * @param taskInstanceId   Existent task id
+     * @param newExecutionTime New execution date
+     * @return void
+     * @see java.time.Instant
+     * @see com.github.kagkarlsson.scheduler.task.TaskInstanceId
+     */
     void reschedule(TaskInstanceId taskInstanceId, Instant newExecutionTime);
 
+    /**
+     * Schedule a new execution date and change the task data to a already existent task
+     *
+     * @param taskInstanceId   Existent task id
+     * @param newExecutionTime New execution date
+     * @param newData          Task instance data
+     * @return void
+     * @see java.time.Instant
+     * @see com.github.kagkarlsson.scheduler.task.TaskInstanceId
+     */
     <T> void reschedule(TaskInstanceId taskInstanceId, Instant newExecutionTime, T newData);
 
+    /**
+     * Cancels a task execution
+     *
+     * @param taskInstanceId Existent task id
+     * @return void
+     * @see com.github.kagkarlsson.scheduler.task.TaskInstanceId
+     */
     void cancel(TaskInstanceId taskInstanceId);
 
+    /**
+     * Get the execution details
```

Comment:
Add:  Uses `Consumer` to avoid forcing the `SchedulerClient` to load all executions in memory